lib/sedna/service/server/base.py

```diff
@@ -50,22 +73,38 @@ def __init__(
         self.url = f"{protocal}://{self.host}:{self.http_port}"
 
     def run(self, app, **kwargs):
-        app.add_middleware(
-            CORSMiddleware, allow_origins=["*"], allow_credentials=True,
-            allow_methods=["*"], allow_headers=["*"],
-        )
+        if hasattr(app, "add_middleware"):
+            app.add_middleware(
+                CORSMiddleware, allow_origins=["*"], allow_credentials=True,
+                allow_methods=["*"], allow_headers=["*"],
+            )
 
         LOGGER.info(f"Start {self.server_name} server over {self.url}")
 
-        uvicorn.run(
+        config = uvicorn.Config(
             app,
             host=self.host,
             port=self.http_port,
             ssl_keyfile=self.keyfile,
             ssl_certfile=self.certfile,
             workers=self.workers,
             timeout_keep_alive=self.timeout,
+            log_level="info",
             **kwargs)
+        server = Server(config=config)
+        with server.run_in_thread():
+            loop = asyncio.get_event_loop()
```

Comment:
Since here create a simple async task, and wait for it, a question: just call wait_stop can also get same effect and be simpler

Reply:
`wait_stop` will wait for a clear shutdown command from the future. When we call `server.run`, it is not expected to be blocked, so that operations after startup can not be executed.

Reply:
fix